src/rtc.rs

```diff
@@ -0,0 +1,448 @@
+    }
+
+    fn get_datetime(&mut self) -> Result<NaiveDateTime, Self::Error> {
+        self.set_24h_fmt();
+
+        let day = self.get_day().unwrap();
+        let month = self.get_month().unwrap();
+        let year = self.get_year().unwrap();
+
+        let seconds = self.get_seconds().unwrap();
+        let minutes = self.get_minutes().unwrap();
+        let hours = hours_to_u8(self.get_hours().unwrap());
+
+        Ok(
+            NaiveDate::from_ymd(year.into(), month.into(), day.into()).and_hms(
+                hours.into(),
+                minutes.into(),
+                seconds.into(),
+            ),
+        )
+    }
+}
+
+// Two 32-bit registers (RTC_TR and RTC_DR) contain the seconds, minutes, hours (12- or 24-hour format), day (day
+// of week), date (day of month), month, and year, expressed in binary coded decimal format
+// (BCD). The sub-seconds value is also available in binary format.
+//
+// The following helper functions encode into BCD format from integer and
+// decode to an integer from a BCD value respectively.
+fn bcd2_encode(word: u32) -> (u8, u8) {
+    (
+        (word / 10).try_into().unwrap(),
+        (word % 10).try_into().unwrap(),
+    )
+}
+
+fn bcd2_decode(fst: u8, snd: u8) -> u32 {
+    (fst * 10 + snd).into()
+}
+
+fn hours_to_register(hours: Hours) -> Result<(u8, u8), Error> {
+    match hours {
+        Hours::H24(h) => Ok(bcd2_encode(h as u32)),
+        Hours::AM(h) => Ok(bcd2_encode((h - 1) as u32)),
+        Hours::PM(h) => Ok(bcd2_encode((h + 11) as u32)),
+    }
+}
+
+fn hours_to_u8(hours: rtcc::Hours) -> u8 {
+    if let rtcc::Hours::H24(h) = hours {
+        h
+    } else {
+        panic!("hours could not be destructured into rtc::Hours::H24(h)");
+    }
+}
+
+/// Enable the low frequency external oscillator. This is the only mode currently
+/// supported, to avoid
+fn enable_lse(bdcr: &mut BDCR, bypass: bool) {
+    bdcr.bdcr()
+        .modify(|_, w| w.lseon().set_bit().lsebyp().bit(bypass));
+    while bdcr.bdcr().read().lserdy().bit_is_clear() {}
+}
+
+fn unlock(apb1: &mut APB1) {
+    let pwr = unsafe { &(*PWR::ptr()) };
+    apb1.enr().modify(|_, w| {
+        w
+            // Enable the backup interface by setting PWREN
+            .pwren()
+            .set_bit()
+    });
+    pwr.cr.modify(|_, w| {
+        w
+            // Enable access to the backup registers
+            .dbp()
+            .set_bit()
+    });
+
+    while pwr.cr.read().dbp().bit_is_clear() {}
+}
+
+fn enable(bdcr: &mut BDCR) {
+    bdcr.bdcr().modify(|_, w| {
+        w
+            // RTC Backup Domain reset bit set high
+            .bdrst()
+            .set_bit()
+    });
+
+    bdcr.bdcr().modify(|_, w| {
+        w
+            // RTC clock source selection
+            .rtcsel()
+            .bits(LSE_BITS)
+            // Enable RTC
+            .rtcen()
+            .set_bit()
+            // RTC backup Domain reset bit set low
+            .bdrst()
+            .clear_bit()
+    });
```

Comment:
The BDCR fields are documented in the PAC, which means they have convenience functions that can make this more readable:

```rust
    bdcr.bdcr().modify(|_, w| w.bdrst().enabled());
    bdcr.bdcr().modify(|_, w| {
        w.rtcsel().lse();
        w.rtcen().enabled();
        w.bdrst().disabled()
    });
```

It's a good idea to use those special accessor functions whenever they are available as it helps readability